cpp/src/arrow/dataset/discovery.cc

```diff
@@ -23,6 +23,7 @@
 #include <unordered_set>
 #include <utility>
 #include <vector>
+#include <glob.h>
```

Comment:
I'm not sure this include will work on Windows.

Reply:
I checked it yesterday, and this include won't work on Windows. I'll look for a way of doing the same exploration in Windows